src/lib/commonPropTypes.ts

```diff
@@ -1,26 +1,43 @@
 import * as PropTypes from 'prop-types'
 import { customPropTypes } from './index'
 
-export const styledComponentPropTypes = {
-  styles: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
-  variables: PropTypes.oneOfType([PropTypes.object, PropTypes.func]),
+export interface CreateCommonConfig {
+  animated?: boolean
+  children?: boolean | 'element'
+  as?: boolean
+  className?: boolean
+  content?: boolean | 'shorthand'
+  styled?: boolean
 }
 
-export const animatedComponentPropTypes = {
-  animation: customPropTypes.animation,
-}
-
-export const commonUIComponentPropTypes = {
-  ...styledComponentPropTypes,
-  ...animatedComponentPropTypes,
-  as: customPropTypes.as,
-  className: PropTypes.string,
-}
-
-export const contentComponentPropsTypes = {
-  content: customPropTypes.contentShorthand,
-}
-
-export const childrenComponentPropTypes = {
-  children: PropTypes.node,
+export const createCommon = (config: CreateCommonConfig = {}) => {
+  const {
+    animated = true,
+    as = true,
+    children = true,
+    className = true,
+    content = true,
+    styled = true,
+  } = config
+  return {
+    ...(animated && {
+      animation: customPropTypes.animation,
+    }),
+    ...(as && {
+      as: customPropTypes.as,
+    }),
+    ...(children && {
+      children: children === true ? PropTypes.node : PropTypes.element,
+    }),
+    ...(className && {
+      className: PropTypes.string,
+    }),
+    ...(content && {
+      content: content === true ? customPropTypes.contentShorthand : customPropTypes.itemShorthand,
```

Comment:
I think that `children` and `content` should accept other values:

- `children: false` => `{}`
- `children: 'node'` => `PropTypes.node`
- `children: 'element'` => `PropTypes.element`

- `content: false` => `{}`
- `content: 'content'` => `customPropTypes.contentShorthand`
- `content: 'item'` => `customPropTypes.itemShorthand,`

Reply:
And use as defaults:
- `children: 'node'` => `PropTypes.node`
- `content: 'content'` => `customPropTypes.contentShorthand`

Reply:
Changed, just note for the content, the values available are `boolean | 'node' | 'shorthand'`, as it is aligned more with the renamed `contentShorthand -> nodeContent` custom prop type. Please take a look.